jbpm-services/jbpm-services-ejb/jbpm-services-ejb-timer/src/main/java/org/jbpm/services/ejb/timer/EJBTimerScheduler.java

```diff
@@ -127,15 +127,20 @@ private void executeTimerJobInstance(TimerJobInstance timerJobInstance) throws E
         try {
             ((Callable<?>) timerJobInstance).call();
         } catch (Exception e) {
-            logger.warn("Execution of time failed due to {}", e.getMessage(), e);
             throw e;
         }
     }
 
     private void recoverTimerJobInstance(EjbTimerJob ejbTimerJob, Exception e) {
-        // if we have next date fired means that it would have been reescheduled already by DefaultTimerJobInstance
         if (ejbTimerJob.getTimerJobInstance().getTrigger().hasNextFireTime() != null) {
-            logger.warn("Execution of time failed Interval Trigger failed {}", ejbTimerJob.getTimerJobInstance());
+            // this is an interval trigger. Problem here is that the timer scheduled by DefaultTimerJobInstance is lost
+            // because of the transaction, so we need to do this here.
+            try {
+                logger.warn("Execution of time failed Interval Trigger failed. Skipping {}", ejbTimerJob.getTimerJobInstance());
+                transaction(job -> internalSchedule(job), ejbTimerJob.getTimerJobInstance());
+            } catch (Exception e1) {
```

Comment:
perhaps, exception message may also be logged in the warning

Reply:
done